OKTA-703920 - ITP event documentation (#4705)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/api/itp-et/index copy.md

| Key Event Attributes  | Description                                         | Data Type      | Example Values |
| --------------------- | --------------------------------------------------- | -------------- | -------------- |
| **event.system.debugContext.debugData**                |                 |         |         |
| ???              |                   | String         | String         |
| ???             |                  | String         | String         |
| **target** (User)         |             |      |        |
| Type        | The type of target object     | String     | User       |
| **actor**                 |  the entity reporting the user risk change (can be a system principal or a user)                 |         |         |
| Type        | The type of actor object           |      |        |
| **client**                |  ???                |       |         |
| IPAddress              | IP address                |       |         |

### device.signals.status.timeout

> **Description:** A registered device associated with at least one user session hasn't communicated with Okta within the required time interval. Use this event to investigate a potentially insecure device and compromised user session. This event contains the device unique identifier in the System Log actor object. You can use this information to find other related events.

Reviewer: ```suggestion
**Description:** This event is fired when a registered device that associated with at least one user session hasn't communicated with Okta within the required time interval. Use this event to investigate a potentially insecure device and compromised user session. This event contains the device unique identifier in the System Log actor object. You can use this information to find other related events.
```

| **target.DetailEntry** (Rule)       |             |      |        |
| PolicyAction        |             | Enum     | `TERMINATE_SESSION`       |
| PolicySingleLogOutEnabled        |             | Boolean     | `true`      |
| PolicySingleLogOutSelectionMode        |            | Enum     | `ALL`       |
| DisplayName        | Display name of the target           | String     |        |
| ID        | Unique identifier of the target            | String     | `00u8xst93qEWYx65sx1d7`       |
| Type        | The type of target object     | String     | PolicyAction      |
| **actor**                 |  ???                 |         |         |
| alternateId        |  Email address of the actor          | String     |        |
| DetailEntry         |             | string     |        |
| DisplayName        | Display name of the actor            | String     |        |
| ID        | Unique identifier of the actor            |      |        |
| Type        | The type of actor object           |      |        |
| **client**                |  ???                |       |         |
| IPAddress              | IP address                |       |         |

### user.session.context.changed

> **Description:** User session context changed. This event indicates that the context in which the session is being used has changed significantly enough from the context in which the event was created, that re-evaluation of policy may be required. Often this indicates a security issue related to the session.

Reviewer: ```suggestion
**Description:** This event indicates that the current session context has changed significantly from the session context when the event was created. This can indicate a security issue related to the session.
```

| Type        | The type of target object     | String     | User       |
| **target** (Session)         |            |      |        |
| alternateId        | Email address of the target           | String     |        |
| DetailEntry         |             |      |        |
| DisplayName        | Display name of the target           | String     |        |
| ID        | Unique identifier of the target            | String     | 00u8xut93qEWYx5sx1d7       |
| Type        | The type of target object     | String     | Session      |
| **actor**                 |  ???                 |         |         |
| alternateId        |  Email address of the actor          | String     |        |
| DetailEntry         |             | string     |        |
| DisplayName        | Display name of the actor            | String     |        |
| ID        | Unique identifier of the actor            |      |        |
| Type        | The type of actor object           |      |        |
| **client**                |  ???                |       |         |
| IPAddress              | IP address                |       |         |

### policy.entity_risk.evaluate

> **Description:** Evaluation of Entity Risk policy. Signal that entity risk policy has been evaluated for an entity for which we have received a risk change event. Event fired when entity risk policy has been evaluated for an entity for which a risk change event was generated.

Reviewer: ```suggestion
**Description:** This event fires when Okta receives a risk event and then evaluates the entity risk policy.
```

| alternateId        | Email address of the target           | String     |        |
| DetailEntry.PolicyAction         |             |  ENUM    | `TERMINATE_ALL_SESSIONS`      |
| DetailEntry.PolicyWorkflowId         |             |  ENUM    |       |
| DisplayName        | Name of the policy action         | String     |`TERMINATE_ALL_SESSIONS`         |
| ID        | Unique identifier of the target            | String     | `00u8xut93qEWYx5sx1d7`       |
| Type        | The type of target object     | String     | PolicyAction      |
| **actor**                 |  ???                 |         |         |
| alternateId        |  Email address of the actor          | String     |        |
| DetailEntry         |             | string     |        |
| DisplayName        | Display name of the actor            | String     |        |
| ID        | Unique identifier of the actor            |      |        |
| Type        | The type of actor object           |      |        |
| **client**                |  ???                |       |         |
| IPAddress              | IP address                |       |         |

### user.session.end

> **Description:** This event is triggered when Okta terminates all IDX sessions. If there are 'N' active sessions for a user the event appears 'N' times. All 'N' events contain `externalSessionId` and `System.Transaction.ID` that correlate with the `System.Transaction.ID` under the `user.session.clear` event.

Reviewer: ```suggestion
**Description:** This event is fired when Okta terminates all IDX sessions for a user. A separate event is logged for each of the user's active sessions. Each event contains `externalSessionId` and `System.Transaction.ID` values that correlate with the `System.Transaction.ID` for the `user.session.clear` event.
```

| ExternalSessionId              |                  | String         | String         |
| **target** (User)         | The user associated with a risk activity            |      |        |
| alternateId        | Email address of the target           | String     |        |
| DetailEntry         |             |      |        |
| DisplayName        | Display name of the target           | String     |        |
| ID        | Unique identifier of the target            | String     | 00u8xut93qEWYx5sx1d7       |
| Type        | The type of target object     | String     | User       |
| **actor**                 |  ???                 |         |         |
| alternateId        |  Email address of the actor          | String     |        |
| DetailEntry         |             | string     |        |
| DisplayName        | Display name of the actor            | String     |        |
| ID        | Unique identifier of the actor            |      |        |
| Type        | The type of actor object           |      |        |
| **client**                |  ???                |       |         |
| IPAddress              | IP address                |       |         |

### user.session.clear

> **Description:**   This event is triggered when an Admin invokes clear sessions by clicking on the "Clear user session". This event appears only once and contains `externalSessionId` and `System.Transaction.ID`.

Reviewer: ```suggestion
**Description:**  This event fires when an admin clears a user's sessions from the user profile. This event appears only once and contains `externalSessionId` and `System.Transaction.ID`.
```

Reviewer: Only once per session? per user?

Author: Once per user session

| **target** (User)         | The user who had their session cleared            |      |        |
| alternateId        | Email address of the target           | String     |        |
| DetailEntry         |             |      |        |
| DisplayName        | Display name of the target           | String     |        |
| ID        | Unique identifier of the target            | String     | 00u8xut93qEWYx5sx1d7       |
| Type        | The type of target object     | String     | User       |
| **actor**                 |  ???                 |         |         |
| alternateId        |  Email address of the actor          | String     |        |
| DetailEntry         |             | string     |        |
| DisplayName        | Display name of the actor            | String     |        |
| ID        | Unique identifier of the actor            |      |        |
| Type        | The type of actor object           |      |        |
| **client**                |  ???                |       |         |
| IPAddress              | IP address                |       |         |

### user.authentication.universal_logout

> **Description:** This event is triggered when an admin or system account triggers Universal Logout against an app instance. It contains the app instance details for which the Universal Logout API was fired. This event identifies when applications have had Universal Logout triggered for audit or debugging purposes. This event is only fired once. It's only fired for applications that have been configured for Universal Logout. You can configure it under Risk policy, Continuous Access policy, or in an admin-initiated Clear User Session.

Reviewer: ```suggestion
**Description:** This event fires when Okta or an admin invokes Universal Logout against an app instance. It contains the app instance details for which the Universal Logout API was fired. This event indicates when apps have had Universal Logout triggered for audit or debugging purposes. This event is only fired once. It's only fired for apps that have been configured for Universal Logout. You can configure it in an entity risk policy or Continuous Access policy, or invoke it manually from the user profile.
```

| **target** (User)         | The user ???           |      |        |
| alternateId        | Email address of the target           | String     |        |
| DetailEntry         |             |      |        |
| DisplayName        | Display name of the target           | String     |        |
| ID        | Unique identifier of the target            | String     | 00u8xut93qEWYx5sx1d7       |
| Type        | The type of target object     | String     | User       |
| **actor**                 |  ???                 |         |         |
| alternateId        |  Email address of the actor          | String     |        |
| DetailEntry         |             | string     |        |
| DisplayName        | Display name of the actor            | String     |        |
| ID        | Unique identifier of the actor            |      |        |
| Type        | The type of actor object           |      |        |
| **client**                |  ???                |       |         |
| IPAddress              | IP address                |       |         |

### workflows.users.delegatedflow.run

> **Description:** This event can be used by admins or security team members to monitor the execution of delegated flows in the Workflows platform from the Admin application. The actor field provides the Okta User ID of the user that ran the flow.  The target fields provide context on the Workflows instance as well as the name and flow id of the executed flow. This event only indicates if the flow was successfully triggered and does not provide information about whether the flow encountered an error.

Reviewer: ```suggestion
**Description:** This event can be used by admins or security team members to monitor the execution of delegated flows in the Workflows platform from the Admin Console. The actor field provides the Okta User ID of the user that ran the flow.  The target fields provide context on the Workflows instance as well as the name and flow ID of the executed flow. This event only indicates if the flow was successfully triggered and doesn't provide information about whether the flow encountered an error.
```
